src/jsonrpc/api.rs

```
@@ -136,6 +136,26 @@ fn list_coins(control: &DaemonControl, params: Option<Params>) -> Result<serde_j
     Ok(serde_json::json!(&res))
 }
 
+fn list_addresses(
+    control: &DaemonControl,
+    params: Option<Params>,
+) -> Result<serde_json::Value, Error> {
+    let start_index: Option<u32> = params
+        .as_ref()
+        .and_then(|p| p.get(0, "start_index"))
+        .and_then(|i| i.as_u64())
+        .and_then(|i| i.try_into().ok());
```

We shouldn't silently ignore invalid parameters to `listaddresses`.